### PR TITLE
Introduce basic version of the mockchain.

### PR DESCRIPTION
**Pull request by qnikst**

### CODE REVIEW

File: chain-impl-mockchain/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+          ]
+edition = "2018"
+
+[dependencies]
+serde = "^1.0"
+serde_derive = "^1.0"
+bincode = "^1.0.1"
+cryptoxide = "*"
+quickcheck = { version = "0.7", optional = true }
+chain-core = { path = "../chain-core", features=["property-test-api"]}
```
Comment (NicolasDP) [inline chain-impl-mockchain/Cargo.toml]: this feature should only be enabled in the `dev-dependencies`

File: chain-impl-mockchain/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+
+pub mod mock;
```
Comment (NicolasDP) [inline chain-impl-mockchain/src/lib.rs]: don't call this module `mock`. We already know that we are using the mockchain.

can you split the module actually ?

* error (with Result/Error defined in there);
* key (PublicKey, PrivateKey, Signature...);
* address (Address);
* transactions (TransactionId, Transaction, SignedTransaction, Witness, UtxoPointer, Output...);
* Block (Block, BlockId, SlockId, ...)

thanks

File: chain-impl-mockchain/Cargo.toml (hunk 1 above)
Comment (qnikst) [reply chain-impl-mockchain/Cargo.toml]: Done

File: chain-impl-mockchain/src/lib.rs (hunk 2 above)
Comment (qnikst) [reply chain-impl-mockchain/src/lib.rs]: Done

**Comment by qnikst** [comment]
@NicolasDP please take another look.

**Pull request merged**